src/events/onMemberJoin.ts

```diff
@@ -7,4 +7,19 @@ export const onMemberJoin = async (member: GuildMember) => {
 
   const welcomeMessage = `Welcome <@${member.id}> to the server!`;
   channel.send(welcomeMessage);
+
+  const roleId = '842455998266605610'; // Overextended Member RoleID
+  const role = member.guild.roles.cache.get(roleId);
+
+  if (!role) {
+    console.error(`Role not found: ${roleId}`);
+    return;
+  }
+
+  try {
+    await member.roles.add(role);
+    console.log(`Assigned role '${role.name}' to ${member.displayName}`);
```

Comment:
Should have a check to see if the member already has the role, unless using `add` already does that and just ignores if the role is present?

Reply:
Looked into it, Discord.js handles these scenarios gracefully. No need for additional check and bot won't freak out, will just ignore.